third_party/2and3/typing_extensions.pyi

```diff
@@ -23,6 +25,24 @@ Final: _SpecialForm = ...
 def final(f: _F) -> _F: ...
 Literal: _SpecialForm = ...
 
+# Internal mypy fallback type for all typed dicts (does not exist at runtime)
+class _TypedDict(Mapping[str, object], metaclass=abc.ABCMeta):
+    def copy(self: _T) -> _T: ...
+    # Using NoReturn so that only calls using mypy plugin hook that specialize the signature
+    # can go through.
+    def setdefault(self, k: NoReturn, default: object) -> object: ...
+    # Mypy plugin hook for 'pop' expects that 'default' has a type variable type.
+    def pop(self, k: NoReturn, default: _T = ...) -> object: ...
+    def update(self: _T, __m: _T) -> None: ...
+    if sys.version_info < (3, 0):
+        def has_key(self, k: str) -> bool: ...
+        def viewitems(self) -> ItemsView[str, object]: ...
+        def viewkeys(self) -> KeysView[str]: ...
+        def viewvalues(self) -> ValuesView[object]: ...
+    def __delitem__(self, k: NoReturn) -> None: ...
+
+def TypedDict(typename: str, fields: Dict[str, Type[_T]], total: bool = ...) -> Type[dict]: ...
```

Comment:
Shouldn't it return `Type[_TypedDict]`?

Reply:
Hm, I would say so, but maybe there is a reason why this is bad. @JukkaL, is this intentional?

Reply:
We just discussed this and decided it should be just `object`, because it is a magic special object like `TypeVar`.